### PR TITLE
Vue3: Add TS / legacy TS CLI templates

### PR DESCRIPTION
**Pull request by chakAs3**
Issue:

## What I did
I have added typescript template for Vue Renderer cli . as when you install via npx on vue3-vite it installs js files with old usage of deprecated stuff, I avoided all deprecation to have a clean typescript template, It will help also maintainers by using generated sandbox repo  with typescript 
## How to test

- [x] Is this testable with Jest or Chromatic screenshots? I guess
- [ ] Does this need a new example in the kitchen sink apps? no need 
- [ ] Does this need an update to the documentation? no 

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

File: code/e2e-tests/addon-interactions.spec.ts
Comment (IanVS) [inline code/e2e-tests/addon-interactions.spec.ts:30]: Why make this change?   It doesn't seem related to the rest of the work here.

File: code/renderers/vue3/template/cli/ts/Button.stories.ts
Comment (IanVS) [inline code/renderers/vue3/template/cli/ts/Button.stories.ts:5]: I don't think we should link directly to the 7.0 docs, since it will become out of date eventually, once new versions are released.  Are there big differences between the 6.5 and 7.0 vue docs right now?

**Review by IanVS — COMMENTED** [review]
Thanks for this!  I'm not a vue user so I'll let others chime in on that part of things.  But I did notice that you're using the `satisfies` syntax that is only supported in TS 4.9+.  Users with older versions of TypeScript will still get the JavaScript assets (after https://github.com/storybookjs/storybook/pull/20458 is merged, at least).  Would you like to also create a `ts-legacy` version of these assets, that doesn't use `satisfies`?

File: code/e2e-tests/addon-interactions.spec.ts
Comment (chakAs3) [reply code/e2e-tests/addon-interactions.spec.ts:30]: oh sorry you are right i was just trying to pass the e2e tests but I put it back

File: code/renderers/vue3/template/cli/ts/Button.stories.ts
Comment (chakAs3) [reply code/renderers/vue3/template/cli/ts/Button.stories.ts:5]: yes  there is major differences

Comment (chakAs3) [reply code/renderers/vue3/template/cli/ts/Button.stories.ts:5]: > Thanks for this! I'm not a vue user so I'll let others chime in on that part of things. But I did notice that you're using the `satisfies` syntax that is only supported in TS 4.9+. Users with older versions of TypeScript will still get the JavaScript assets (after #20458 is merged, at least). Would you like to also create a `ts-legacy` version of these assets, that doesn't use `satisfies`?

i will do it I have it already no major changes apart of the satisfies

Comment (IanVS) [reply code/renderers/vue3/template/cli/ts/Button.stories.ts:5]: OK, we may just need to remember to update these links after 7.0 final is released, I guess.

Comment (chakAs3) [reply code/renderers/vue3/template/cli/ts/Button.stories.ts:5]: > I don't think we should link directly to the 7.0 docs, since it will become out of date eventually, once new versions are released. Are there big differences between the 6.5 and 7.0 vue docs right now?

basically this code  is for for 7.0  compatibility starts from this version and if there is a new version requires changes will have to do it, and anyway I'm following the pattern if you look at  renderers/react or svelte they are pointing to docs/7.0

**Comment by chakAs3** [comment]
> Thanks for this! I'm not a vue user so I'll let others chime in on that part of things. But I did notice that you're using the `satisfies` syntax that is only supported in TS 4.9+. Users with older versions of TypeScript will still get the JavaScript assets (after #20458 is merged, at least). Would you like to also create a `ts-legacy` version of these assets, that doesn't use `satisfies`?

it is already done @IanVS 👍   It is late for you now since you are in Michigan, nobody is here to approve my PR 😔

**Review by IanVS — APPROVED** [review]
Looks good to me, thanks @chakAs3.

**Review by shilman — APPROVED** [review]
Great job @chakAs3 and happy new year!

**Comment by chakAs3** [comment]
> Great job @chakAs3 and happy new year!

Thanks @shilman Happy new year from Dubai I hope it will be an amazing year for Everyone,and Storybook will rock

**Pull request merged**
